### PR TITLE
[RISCV] Split out VSETVLIInfo AVL states to be more explicit

### PR DESCRIPTION
**Pull request by lukel97**
We currently use AVLIsReg to represent VLMAX as well as a dummy value for
whenever the VL is ignored by vmv.x.s. This splits them out into separate
states so that AVLIsReg is always a virtual register and should help with
tracking the definition inside VSETVLIInfo directly in #89180.

This is almost an NFC but it sets the kill flag for x0 in more places.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp [hunk 1]
```diff
@@ -935,14 +964,13 @@ static VSETVLIInfo computeInfoForInstr(const MachineInstr &MI, uint64_t TSFlags,
   // AVL operand with the AVL of the defining vsetvli.  We avoid general
   // register AVLs to avoid extending live ranges without being sure we can
   // kill the original source reg entirely.
-  if (InstrInfo.hasAVLReg() && InstrInfo.getAVLReg().isVirtual()) {
-    MachineInstr *DefMI = MRI->getVRegDef(InstrInfo.getAVLReg());
-    if (DefMI && isVectorConfigInstr(*DefMI)) {
+  if (InstrInfo.hasAVLReg()) {
+    MachineInstr *DefMI = MRI->getUniqueVRegDef(InstrInfo.getAVLReg());
+    if (isVectorConfigInstr(*DefMI)) {
```
Comment (BeMg) [inline llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp:969]: getUniqueVRegDef may return a nullptr if none is found, although a valid program should always have a VRegDef here.

Comment (lukel97) [reply llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp:969]: Woops yes, that should have an assert that DefMI isn't null underneath there. And then we can remove it after we move the definitions into VSETVLIInfo.

**Comment by BeMg** [comment]
Inside `bool RISCVInsertVSETVLI::needVSETVLIPHI`

Maybe the following statement could be dropped due to AVLReg must be Virtual now.
```
  Register AVLReg = Require.getAVLReg();
  if (!AVLReg.isVirtual())
    return true;
```

**Review by BeMg — APPROVED** [review]
LGTM

**Pull request merged**